src/main/java/org/testng/internal/TestResult.java

```diff
@@ -320,4 +336,90 @@ public boolean wasRetried() {
   public void setWasRetried(boolean wasRetried) {
     this.m_wasRetried = wasRetried;
   }
+
+  public List<ITestNGMethod> getSkipCausedBy() {
+    if (this.m_status != SKIP) {
+      return Lists.newArrayList();
+    }
+    if (skipAnalysed) {
+      return Collections.unmodifiableList(skippedDueTo);
+    }
+    skipAnalysed = true;
+    //check if there were any config failures
+    Set<ITestResult> skippedConfigs = m_context.getFailedConfigurations().getAllResults();
+    for (ITestResult skippedConfig : skippedConfigs) {
+      if (isGlobalFailure(skippedConfig) ||
+          isRelated(skippedConfig) || belongToSameGroup(skippedConfig)) {
+        skippedDueTo.add(skippedConfig.getMethod());
+        return Collections.unmodifiableList(skippedDueTo);
+      }
+    }
+    //Looks like we didnt have any configuration failures. So some upstream method perhaps failed.
+    List<String> upstreamMethods = Arrays.asList(m_method.getMethodsDependedUpon());
+    if (upstreamMethods.isEmpty()) {
+      //Maybe group dependencies exist ?
+      List<String> upstreamGroups = Arrays.asList(m_method.getGroupsDependedUpon());
+      if (upstreamGroups.isEmpty()) {
+        return Collections.unmodifiableList(skippedDueTo);
+      }
+      List<ITestResult> allFailures = Lists.newArrayList(m_context.getFailedTests().getAllResults());
+      allFailures.addAll(m_context.getFailedButWithinSuccessPercentageTests().getAllResults());
+      skippedDueTo = allFailures.stream()
+          .map(ITestResult::getMethod)
+          .filter(each -> {
+            List<String> currentMethodGroups = Arrays.asList(each.getGroups());
+            List<String> interection = Lists.intersection(upstreamGroups, currentMethodGroups);
+            return !interection.isEmpty();
+          }).collect(Collectors.toList());
+
+      return Collections.unmodifiableList(skippedDueTo);
+    }
+
+    //So we have dependsOnMethod failures
+    List<ITestResult> allfailures = Lists.newArrayList(m_context.getFailedTests().getAllResults());
+    allfailures.addAll(m_context.getFailedButWithinSuccessPercentageTests().getAllResults());
+    skippedDueTo = allfailures.stream()
+        .map(ITestResult::getMethod)
+        .filter(each -> upstreamMethods.contains(each.getQualifiedName()))
+        .collect(Collectors.toList());
+    return Collections.unmodifiableList(skippedDueTo);
+  }
+
+  private boolean isGlobalFailure(ITestResult result) {

```

Comment:
static here and later

Reply:
Only this method can be made static. Fixed it.